dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.cpp

```diff
@@ -0,0 +1,372 @@
+    {
+        THROW_HR_IF(E_INVALIDARG, !m_audio.empty());
+
+        m_audio = L"<audio src=\"" + audioUri.ToString() + L"\" loop=\"true\"/>";
+        m_duration = duration;
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetAudioEvent(AppNotificationSoundEvent const& soundEvent)
+    {
+        THROW_HR_IF(E_INVALIDARG, !m_audio.empty());
+
+        // Need to map enums to strings :)
+        m_audio = L"<audio src=\"" + GetWinSoundEventString(soundEvent) + L"\"/>";
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetAudioEvent(AppNotificationSoundEvent const& soundEvent, AppNotificationDuration const& duration)
+    {
+        THROW_HR_IF(E_INVALIDARG, !m_audio.empty());
+
+        // Need to map enums to strings :)
+        m_audio = L"<audio src=\"" + GetWinSoundEventString(soundEvent) + L"\" loop=\"true\"/>";
+        m_duration = duration;
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::MuteAudio()
+    {
+        THROW_HR_IF(E_INVALIDARG, !m_audio.empty());
+
+        m_audio = L"<audio silent=\"true\"/>";
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::AddButton(AppNotificationButton const& value)
+    {
+        THROW_HR_IF(E_INVALIDARG, m_inputList.size() == 5u);
+
+        m_inputList.push_back(value.as<winrt::Windows::Foundation::IStringable>().ToString());
+        m_useButtonStyle = value.ButtonStyle() != AppNotificationButtonStyle::Default;
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetTag(winrt::hstring const& value)
+    {
+        m_tag = value;
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationBuilder AppNotificationBuilder::SetGroup(winrt::hstring const& value)
+    {
+        m_group = value;
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::AppNotification AppNotificationBuilder::BuildNotification()
+    {
+        winrt::hstring xmlResult{ L"<toast" };
+
+        // Add duration attribute if set
+        if (m_duration != AppNotificationDuration::Default)
+        {
+            xmlResult = xmlResult + L" duration=\"long\"";
+        }
+
+        // Add scenario attribute if set
+        if (m_scenario != AppNotificationScenario::Default)
+        {
+            winrt::hstring scenario{};
+            switch (m_scenario)
+            {
+            case AppNotificationScenario::Alarm:
+                scenario = L"\"alarm\"";
```

Comment:
`scenario = LR"("alarm")";`

Raw strings make complex strings easier to read, and R"( before and )" after lets you embed double quotes in a string without escaping them

same comment applies throughout

Reply:
Are you constructing XML? You can use apostraphe to quote arguments instead of double-quotes which simplifies a lot of your string syntax e.g. line 294

`scenario = L"\"reminder\"";`

produces on line 304 XML of `...scenario="reminder"...`

but `scenario = L"'reminder'";` would produce `...scenario='reminder'...`

Reply:
Yes, we are building xml. Will move to use ' intsead of of ".